sql/core/src/test/resources/sql-tests/inputs/cross-join.sql

```diff
@@ -33,3 +33,5 @@ create temporary view D(d, vd) as select * from nt1;
 -- Allowed since cross join with C is explicit
 select * from ((A join B on (a = b)) cross join C) join D on (a = d);
 
+-- Cross joins with non-equal predicates
+SELECT * FROM nt1 CROSS JOIN nt2 ON (nt1.k > nt2.k);
```

Comment:
So far, the SQL syntax allows users to specify the join condition. 

The Dataset API does not allow users to do it, but users still can do it by using the filter, which will be pushed into the cross join.